lib/chipmunk_validatable.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module ChipmunkValidatable
+
+  def self.included(base)
+    base.extend(ClassMethods)
+  end
+
+  module ClassMethods
+    def validators
+      @validators ||= []
+    end
+
+    def validates(_description = "", precondition: ->{}, condition:, error:)
```

Comment:
This looks like black magic to me. My main concern would be incomprehensible stacktraces when there is a problem. If the condition raises an error, what happens?  Is there a missing try/catch here?

Reply:
An exception in the validator would in fact be exceptional - normally it would return false. Exceptional conditions do get caught here: https://github.com/mlibrary/chipmunk/pull/30/files#diff-183075b1a34a924c5f0e3e92dd41cc2aL27

The stack trace at least shows something understandable for the topmost frame.

```     RuntimeError:
       stack trace plz
     # ./lib/chipmunk_bag_validator.rb:14:in `block in <class:ChipmunkBagValidator>'
     # ./lib/chipmunk_validatable.rb:17:in `instance_exec'
     # ./lib/chipmunk_validatable.rb:17:in `block in validates'
     # ./lib/chipmunk_validatable.rb:30:in `instance_exec'
     # ./lib/chipmunk_validatable.rb:30:in `block in valid?'
     # ./lib/chipmunk_validatable.rb:29:in `each'
     # ./lib/chipmunk_validatable.rb:29:in `reduce'
     # ./lib/chipmunk_validatable.rb:29:in `valid?'
     # ./app/jobs/bag_move_job.rb:24:in `perform'
```

Does that seem reasonable?